bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/HistoryStore.java

```diff
@@ -109,4 +109,8 @@ private void writeHistory() {
 		settingsManager.put(sectionName, names);
 	}
 
+	public List<String> asList() {
+		return new ArrayList<>(history);
```

Comment:
What is the intention of this method? Immutable copy? then use List.copyOf()

Reply:
The intention of this method is that I want to have access to methods implemented by List (but really only indexOf and size()).
Now that you question this method, I start having doubts myself whether it's a good idea to have it.
An alternative would be implementing size() and indexOf() in HistoryStore. What do you think?

Reply:
sounds good - avoids the copy

Reply:
I agree - the code became much clearer! https://github.com/eclipse-platform/eclipse.platform.ui/commit/060b9941fea88d89e98acb6b8adea55cde83bbe7